mesonbuild/utils/universal.py

```diff
@@ -1280,11 +1283,15 @@ def get_variable_regex(variable_format: Literal['meson', 'cmake', 'cmake@'] = 'm
         ''', re.VERBOSE)
     else:
         regex = re.compile(r'''
-            (?:\\\\)+(?=\\?\$)  # Match multiple backslashes followed by a dollar sign
+            (?:\\\\)+(?=\\?(\$|@))  # Match multiple backslashes followed by a dollar sign
```

Comment:
looks like the comment should be updated

Reply:
Actually I need to revert this, it was part of the attempt at updating the regex which ultimately proved to be unreasonable

Reply:
reinspecting this with a less tired mind, the change is actually right and I do need to update the comment

Reply:
resolved by 5c758763bf0a8bf63099701dfa96525a7ae7f048